### ISSUE TITLE
TypeError: chunk.isInitial is not a function

### ISSUE THREAD

**Issue by daryn-k** [webpack-contrib/extract-text-webpack-plugin#449]
There is this error:
```
var shouldExtract = !!(options.allChunks || chunk.isInitial());
TypeError: chunk.isInitial is not a function

```
I have no idea what is it.

My webpack.config.js file:

```
var ExtractTextPlugin = require('extract-text-webpack-plugin');

module.exports = {
    entry: './app/client.js',
    output: {
        filename: 'bundle.js',
        path: 'public/js'
    },
    module: {
        loaders: [
            {
                test: /\.jsx?$/,
                exclude: /node_modules/,
                loader: 'babel-loader',
                query: {
                    presets: ['react', 'es2015', 'stage-1']
                }
            },
            {
                test: /\.css/,
                loader: ExtractTextPlugin.extract('css')
            }
        ]
    },
    plugins: [
        new ExtractTextPlugin('styles.css')
    ]
};
```

**Comment by bebraw** [comment]
There's not enough information to debug. Which version of webpack/extract-text-webpack-plugin are you using? If you have webpack 1 and are using extract-text-webpack-plugin 2, that error could happen. In that case you'll have to use extract-text-webpack-plugin 1 or upgrade webpack to 2.

**Comment by daryn-k** [comment]
I use last versions of 'extract-text-webpack-plugin' and webpack (2.2.1)

**Comment by bebraw** [comment]
Ok, still not enough info. Maybe someone else has a better idea.

**Comment by daryn-k** [comment]
I don't know what else... It's  just about a simple functionality - build 'bundle.css' from all css-files from project. What kind of information do you need?

**Comment by bebraw** [comment]
I would need a project to run. `chunk.isInitial()` is something provided by webpack itself. If `chunk` is missing, there is something fatally wrong.

**Comment by daryn-k** [comment]
Maybe these lines?

```
 at C:\site.dev\site-spa\react\node_modules\extract-text-webpack-plugin\index.js:259:55
    at C:\site.dev\site-spa\react\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:3025:16
    at eachOfArrayLike (C:\site.dev\site-spa\react\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:940:9)
    at eachOf (C:\site.dev\site-spa\react\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:990:5)
    at Object.eachLimit (C:\site.dev\site-spa\react\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:3089:3)
    at ExtractTextPlugin.<anonymous> (C:\site.dev\site-spa\react\node_modules\extract-text-webpack-plugin\index.js:257:10)
    at Compilation.applyPluginsAsync (C:\site.dev\site-spa\react\node_modules\tapable\lib\Tapable.js:71:13)
    at Compilation.seal (C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:525:7)
    at Compiler.<anonymous> (C:\site.dev\site-spa\react\node_modules\webpack\lib\Compiler.js:397:15)
    at C:\site.dev\site-spa\react\node_modules\tapable\lib\Tapable.js:103:11
    at Compilation.<anonymous> (C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:445:10)
    at C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:417:12
    at C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:332:10
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:52:16
    at done (C:\site.dev\site-spa\react\node_modules\async\lib\async.js:246:17)
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:44:16
    at C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:332:10
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:52:16
    at done (C:\site.dev\site-spa\react\node_modules\async\lib\async.js:246:17)
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:44:16
    at C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:332:10
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:52:16
    at done (C:\site.dev\site-spa\react\node_modules\async\lib\async.js:246:17)
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:44:16
    at C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:332:10
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:52:16
    at done (C:\site.dev\site-spa\react\node_modules\async\lib\async.js:246:17)
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:44:16
    at C:\site.dev\site-spa\react\node_modules\webpack\lib\Compilation.js:332:10
    at C:\site.dev\site-spa\react\node_modules\async\lib\async.js:52:16
```

**Comment by bebraw** [comment]
Sorry, that doesn't help.

I'm closing this issue until someone can provide a proper project to examine as instructed in the issue template.

**Comment by Rob--** [comment]
Unable to provide a project, but I'd just like to mention that this is happening to me also.

Package versions:
```
"webpack": "^1.14.0"
"extract-text-webpack-plugin": "^2.1.0"
```

Properties from config file:
```
module: {
  rules: [
    {
      test: /\.css$/,
      use: ExtractTextPlugin.extract({
        fallback: "style-loader",
        use: "css-loader"
      })
    }
  ]
},
plugins: [
  new ExtractTextPlugin("styles.css")
]
```

**Comment by bebraw** [comment]
@Rob-- Can you either upgrade to webpack 2 or downgrade the plugin to 1 series? I don't think the 2.x version of the plugin is supposed to work with webpack 1. That could be forced through a peer dependency.

**Comment by Rob--** [comment]
@bebraw I'm a complete idiot, for some reason I thought I was using Webpack 2.0 even though I saw my Webpack version was clearly not 2.0. It's working perfectly now, sorry for wasting your time!

**Comment by bebraw** [comment]
@Rob-- No probs. We may still want to set that peer dep to avoid these kind of issues. 👍

**Comment by jony89** [comment]
occurs for me as well - same exact issue same exact config.

webpack : 2.2.1 
extract-text-webpack-plugin : 2.1.0

**Comment by bebraw** [comment]
@jony89 There's still not enough info to help. 😢

**Comment by jony89** [comment]
@bebraw what do you need?

latest webpack, latest package of this module. add the following config to webpack : 

```js
        rules: [
            {
                test: /\.css$/, 
                loader: ExtractTextPlugin.extract({
                   fallback: 'style-loader',
                   use: [ 
                     { 
                       loader: 'css-loader', 
                       options : { autoprefixer: false, sourceMap: true, importLoaders: 1 } 
                     }, 
                     'postcss-loader'
                   ]
                 })
            }            
        ],
```

and : 

```js
    plugins: [
        new ExtractTextPlugin('index.css'),
    ]
```

Then run the project with that webpack config and you will get : 

> node_modules\extract-text-webpack-plugin\index.js:267
> 				var shouldExtract = !!(options.allChunks || chunk.isInitial());
> 				                                                  ^
> 
> TypeError: chunk.isInitial is not a function
>     at C:\chikichak\app\client\chikichak-mobile\node_modules\extract-text-webpack-plugin\index.js:267:55
>     at C:\chikichak\app\client\chikichak-mobile\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:3025:16
>     at eachOfArrayLike (C:\chikichak\app\client\chikichak-mobile\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:940:9)
>     at eachOf (C:\chikichak\app\client\chikichak-mobile\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:990:5)
>     at Object.eachLimit (C:\chikichak\app\client\chikichak-mobile\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:3089:3)
>     at ExtractTextPlugin.<anonymous> (C:\chikichak\app\client\chikichak-mobile\node_modules\extract-text-webpack-plugin\index.js:265:10)
>     at Compilation.applyPluginsAsync (C:\chikichak\app\client\chikichak-mobile\node_modules\gulp-webpack\node_modules\tapable\lib\Tapable.js:71:13)
>     at Compilation.seal (C:\chikichak\app\client\chikichak-mobile\node_modules\gulp-webpack\node_modules\webpack\lib\Compilation.js:525:7)
>     at Compiler.<anonymous> (C:\chikichak\app\client\chikichak-mobile\node_modules\gulp-webpack\node_modules\webpack\lib\Compiler.js:397:15)
>     at C:\chikichak\app\client\chikichak-mobile\node_modules\gulp-webpack\node_modules\tapable\lib\Tapable.js:103:11
>     at Compilation.<anonymous> (C:\chikichak\app\client\chikichak-mobile\node_modules\gulp-webpack\node_modules\webpack\lib\Compilation.js:445:10)
>     at C:\chikichak\app\client\chikichak-mobile\node_modules\gulp-webpack\node_modules\webpack\lib\Compilation.js:417:12
>     at C:\chikichak\app\client\chikichak-mobile\node_modules\gulp-webpack\node_modules\webpack\lib\Compilation.js:332:10
>     at C:\chikichak\app\client\chikichak-mobile\node_modules\async\lib\async.js:52:16
>     at done (C:\chikichak\app\client\chikichak-mobile\node_modules\async\lib\async.js:246:17

**Comment by bebraw** [comment]
@jony89 I would need a standalone project. Any chance you could open a separate issue with that?

**Comment by jony89** [comment]
k

**Comment by jony89** [comment]
see #461

**Comment by bebraw** [comment]
@jony89 Ok. Hopefully someone can debug based on that. Thanks.

**Comment by michael-ciniawsky** [comment]
@jony89 There is no `{ autoprefixer: false }` in `css-loader`, which version of `css-loader` do you use and does it work without setting `sourceMap`. Please also try without `postcss-loader` and provide your `postcss.config.js` aswell. Don't take it personal but I won't download any kind of `.zip` or other files as long as I cannot take a first glance at them before I execute that on my system 😛 (#461)

**Comment by jony89** [comment]
so how do you expect me to provide a stand alone project?

create one by your own, it is pretty easy. postcss is not needed to reproduce the issue. (I just tried to create standalone as fast as possible) this will trigger the same issue : 

```js
        rules: [
            {
                test: /\.css$/, 
                loader: ExtractTextPlugin.extract({
                   fallback: 'style-loader',
                   use: [ 
                     { 
                       loader: 'css-loader'
                     }
                   ]
                 })
            }            
        ],

```

It start to look like you guys don't really want to try and fix that. I'm trying to help here.

**Comment by Rob--** [comment]
@jony89 you could make a new repo and upload an example project there?

**Comment by jony89** [comment]
god bless u all.

https://github.com/jony89/simple-webpack-extract-plugin

**Comment by bebraw** [comment]
@jony89 `node_modules/.bin/webpack --config webpack.dev.config.js` against that completes without a problem for me (Node 6). There's still some info missing (Node version for one). Missing something subtle.

**Comment by 1e1f** [comment]
This is a little old now, but I just ran into this same error when I moved to a machine running node 7.0.0, after installing nvm and switching to 7.10.0 the error was resolved. webpack2.2 / extract 2.1

**Comment by flyon** [comment]
Same problem. All current stable versions: webpack 2.5.1 / extract-text-webpack-plugin 2.1.0. Just upgraded to node 7.10.0, still have the problem.

    TypeError: extractedChunk.isInitial is not a function

**Comment by jony89** [comment]
some of the conversation moved to the issue, so i write my temp solution here as well

tnx, temporary solution is to give webpack-stream your own webpack version: 

```js
webpackStream(config, require('webpack'))
```

see https://github.com/shama/webpack-stream/issues/125

this is the issue : #461 

hope it helps for others.

**Comment by flyon** [comment]
I thought I thoroughly checked all the versions, but it turned out I had a different webpack version running in node_modules of a higher level folder. So webpack was globally 2.5.1 but require('webpack') was an older version. Working fine now that I've got that sorted out. You can close this again

**Comment by elliotwesoff** [comment]
Hello I am getting this error as well.
```
$ webpack -v
3.0.0
$ node -v
v7.7.1
```
```
 optimizeUnhandled rejection TypeError: chunk.isInitial is not a function
    at C:\project\node_modules\extract-text-webpack-plugin\index.js:267:55
    at C:\project\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:3047:16
    at eachOfArrayLike (C:\project\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:1039:9)
    at eachOf (C:\project\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:1087:5)
    at Object.eachLimit (C:\project\node_modules\extract-text-webpack-plugin\node_modules\async\dist\async.js:3109:3)
    at ExtractTextPlugin.<anonymous> (C:\project\node_modules\extract-text-webpack-plugin\index.js:265:10)
    at Compilation.applyPluginsAsync (C:\project\node_modules\tapable\lib\Tapable.js:71:13)
    at Compilation.seal (C:\project\node_modules\webpack\lib\Compilation.js:525:7)
    at Compiler.<anonymous> (C:\project\node_modules\webpack\lib\Compiler.js:397:15)
    at C:\project\node_modules\tapable\lib\Tapable.js:103:11
    at C:\project\node_modules\html-webpack-plugin\index.js:60:9
    at tryCatcher (C:\project\node_modules\html-webpack-plugin\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\project\node_modules\html-webpack-plugin\node_modules\bluebird\js\release\promise.js:512:31)
    at Promise._settlePromise (C:\project\node_modules\html-webpack-plugin\node_modules\bluebird\js\release\promise.js:569:18)
    at Promise._settlePromise0 (C:\project\node_modules\html-webpack-plugin\node_modules\bluebird\js\release\promise.js:614:10)
    at Promise._settlePromises (C:\project\node_modules\html-webpack-plugin\node_modules\bluebird\js\release\promise.js:693:18)
````

**Comment by ghostd** [comment]
@elliotwesoff look at this [html-webpack-plugin comment](https://github.com/jantimon/html-webpack-plugin/issues/705#issuecomment-309719976), it fixed my 'undefined' error (mainly back to webpack@2.6.1 or use npm@5)

**Comment by elliotwesoff** [comment]
Hey @ghostd , I still get the same error after doing `cd node_modules/html-webpack-plugin && npm install webpack extract-text-webpack-plugin`

**Comment by joehannes** [comment]
same issue here. jony89 [comment](https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/449#issuecomment-303015864) helped me in my case, even though i'm using gulp-webpack, not webpack-stream as he states.
I got my [setup](http://www.github.com/joehannes-jobs/webtrekk) on github, so you can clone and play around if you want to ...

Just to let you know:
gulp-webpack 1.5.0
webpack 2.6.1
extract-text-webpack-plugin 2.1.2
sass-loader (which is using E-T-W-P) 6.0.6

Basically, to make it work, try and provide to your webpack-stream or as in my case gulp-webpack webpack as a second param after your config ... that's what makes it work in my case as well :)

**Comment by iddqdidkfa** [comment]
Can't believe this issue still closed:

node: 8.1.4
webpack: 3.1.0
extract-text-webpack-plugin: 3.0

**Comment by zullusa** [comment]
Hi! Can You review this [article](https://habrahabr.ru/post/309958/)? You even don't need to know russian. Just execute commands and create files. I get topic error, when I execute `npm run build`.
TypeError: chunk.sortModules is not a function. 

What has forgotten the article author to do?

**Comment by marcopeg** [comment]
Hi guys, I ran into this problem myself. After reading it I tried to downgrade the plugin to 2.1.0 and it works for me. So far I can say that 3.x didn't work with webpack 2.2.0.

Hope this will help somebody here ;-)

**Comment by nfriend** [comment]
Thanks @marcopeg, I had the same issue.  For me I was able to instead fix this issue by upgrading from webpack 2 to webpack 3.

**Comment by jk171505** [comment]
node: 8.4.0
webpack: 3.0.0, 3.1.0, 3.5.5
extract-text-webpack-plugin: 3.0

the same problem...

**Comment by sundagy** [comment]
I am install local version webpack-dev-server and it solved the problem.

**Comment by brpsmith** [comment]
I solved this issue by upgrading webpack-stream.  

webpack 3.5.6
webpack-stream 4.0.0
extract-text-webpack-plugin 3.0.0

**Comment by enmanuelduran** [comment]
Im having this situation with webpack 1.13.2 and extract-text-webpack-plugin 3.0.1 :/

**Comment by navasiloy** [comment]
# for webpack 3 
npm install --save-dev extract-text-webpack-plugin
# for webpack 2 
npm install --save-dev extract-text-webpack-plugin@2.1.2
# for webpack 1 
npm install --save-dev extract-text-webpack-plugin@1.0.1

**Comment by enmanuelduran** [comment]
@navasiloy your suggestion made the trick for my case. thanks.

**Comment by andrew-aladev** [comment]
> I'm closing this issue until someone can provide a proper project to examine as instructed in the issue template.

@webpack You definitely need another admin for this repository.

**Comment by evilebottnawi** [comment]
@andrew-aladev please provide minimum reproducible test repo

**Comment by andrew-aladev** [comment]
I was able to reproduce it with empty string as an extension for `webpacker`. I am going to provide more detailed info later.

**Comment by evilebottnawi** [comment]
@andrew-aladev just ping me when repo is ready :+1:

**Comment by andrew-aladev** [comment]
My chunks was created in `webpack/lib/Compilation.js` in `addChunk`. But my version of webpack is `1.15.0`. So we have misconfigured webpack here.

It is an npm issue. It shouldn't be possible for me to install latest `extract-text-webpack-plugin` with webpack 1x.

**Comment by evilebottnawi** [comment]
latest `extract-text-webpack-plugin` is not compatibility with `webpack@1`

**Comment by andrew-aladev** [comment]
The problem was about `webpacker`. It was using the wrong version of `webpack`. It's so easy to receive the same problem with npm.
